x-pack/plugins/task_manager/server/metrics/counter/metric_counter_service.ts

```diff
@@ -0,0 +1,78 @@
+/*
```

Comment:
Previously I had a `SuccessRateCounter` that was very tied to counting successful attempts vs total attempts. This was too restrictive so this `MetricCounterService` gives more flexibility.